tests/nox/test_git_nox.py

```diff
@@ -0,0 +1,67 @@
+from unittest import mock
+
+import pytest
+from git import Repo
+
+from noxfiles.git_nox import generate_tag
+
+
+class TestGitNox:
+    """
+    Tests for git nox commands and/or utilities
+    """
+
+    @pytest.fixture(scope="session")
+    def repo(self):
+        repo = Repo()
+        git_session = repo.git()
+        git_session.fetch("--force", "--tags")
+        return repo
+
+    @mock.patch("nox.Session.log")
+    @pytest.mark.parametrize(
+        "current_branch,tags,expected_tag",
+        [
+            ("main", ["2.9.0"], "2.9.1b0"),
+            ("main", ["2.9.1", "2.9.0"], "2.9.2b0"),
+            ("some-test-feature-branch", ["2.9.1"], "2.9.2a0"),
+            ("some-test-feature-branch", ["2.9.1", "2.9.0"], "2.9.2a0"),
+            (
+                "some-test-feature-branch",
+                ["2.9.3a0", "2.9.2"],
+                "2.9.3a1",
+            ),  # our repo happens to already have a 2.9.3a0 tag, let's use it for testing here
+            ("release-2.9.0", [], "2.9.0rc0"),
+            ("release-2.9.0", ["2.9.0rc1"], "2.9.0rc2"),
+            (
+                "release-2.9.0",
+                ["2.9.1"],
+                "2.9.0rc0",
+            ),  # even with a release further ahead, the RC tag generation relies on the release branch name
+            # release branches MUST adhere to `release-n.n.n` convention
+            # if not, they will be treated as any other "feature" branch
+            ("release/2.9.0", ["2.9.0"], "2.9.1a0"),
+            ("release_2.9.0", ["2.9.0"], "2.9.1a0"),
```

Comment:
nit: Super polish suggestion - if you detect the word `release` in the branch name, throw a warning like:
```
WARNING: Current branch name 'release/testing' does not match release branch format ('release-n.n.n'), so will be detected as a feature branch instead. Did you mean to name your branch differently?
```

Reply:
nice, very defensive, i like it. i'll add. thanks!